Source/Diagnostics/BoostedFrameDiagnostic.cpp

```diff
@@ -1416,52 +1428,106 @@ LabFrameSlice::
 AddPartDataToParticleBuffer(
     Vector<WarpXParticleContainer::DiagnosticParticleData> tmp_particle_buffer,
     int nSpeciesBoostedFrame) {
+
+
     for (int isp = 0; isp < nSpeciesBoostedFrame; ++isp) {
         auto np = tmp_particle_buffer[isp].GetRealData(DiagIdx::w).size();
 
         if (np == 0) return;
 
-        auto const& wpc = tmp_particle_buffer[isp].GetRealData(DiagIdx::w);
-        auto const& xpc = tmp_particle_buffer[isp].GetRealData(DiagIdx::x);
-        auto const& ypc = tmp_particle_buffer[isp].GetRealData(DiagIdx::y);
-        auto const& zpc = tmp_particle_buffer[isp].GetRealData(DiagIdx::z);
-        auto const& uxpc = tmp_particle_buffer[isp].GetRealData(DiagIdx::ux);
-        auto const& uypc = tmp_particle_buffer[isp].GetRealData(DiagIdx::uy);
-        auto const& uzpc = tmp_particle_buffer[isp].GetRealData(DiagIdx::uz);
-
-        particles_buffer_[isp].resize(np);
-        auto wpc_buff = particles_buffer_[isp].GetRealData(DiagIdx::w);
-        auto xpc_buff = particles_buffer_[isp].GetRealData(DiagIdx::x);
-        auto ypc_buff = particles_buffer_[isp].GetRealData(DiagIdx::y);
-        auto zpc_buff = particles_buffer_[isp].GetRealData(DiagIdx::z);
-        auto uxpc_buff = particles_buffer_[isp].GetRealData(DiagIdx::ux);
-        auto uypc_buff = particles_buffer_[isp].GetRealData(DiagIdx::uy);
-        auto uzpc_buff = particles_buffer_[isp].GetRealData(DiagIdx::uz);
-
-
-        int partcounter = 0;
-        for (int i = 0; i < np; ++i)
-        {
-           if( xpc[i] >= (diag_domain_lab_.lo(0)-dx_) &&
-               xpc[i] <= (diag_domain_lab_.hi(0)+dx_) ) {
- #if (AMREX_SPACEDIM == 3)
-              if( ypc[i] >= (diag_domain_lab_.lo(1)-dy_) &&
-                  ypc[i] <= (diag_domain_lab_.hi(1) + dy_))
- #endif
-              {
-                 wpc_buff[partcounter] = wpc[i];
-                 xpc_buff[partcounter] = xpc[i];
-                 ypc_buff[partcounter] = ypc[i];
-                 zpc_buff[partcounter] = zpc[i];
-                 uxpc_buff[partcounter] = uxpc[i];
-                 uypc_buff[partcounter] = uypc[i];
-                 uzpc_buff[partcounter] = uzpc[i];
-                 ++partcounter;
-              }
-           }
-        }
+        Real* const AMREX_RESTRICT wp_temp =
```

Comment:
Same thing, couldn't all the `_temp` variables be `Real const * const` instead of `Real * const`?

Reply:
Yes. made this change now.